fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/MatchCandidate.java

```diff
@@ -314,7 +331,7 @@ static Optional<MatchCandidate> fromPrimaryDefinition(@Nonnull final RecordMetaD
                                                           final boolean isReverse) {
         if (commonPrimaryKey != null) {
             final var availableRecordTypes = metaData.getRecordTypes().keySet();
-            final var baseRef = createBaseRef(metaData, availableRecordTypes, recordTypes);
+            final var baseRef = createBaseRef(metaData, availableRecordTypes, recordTypes, commonPrimaryKey.toString());
```

Comment:
I think this is wrong, but don't know how to represent name of the primary key?

Reply:
use empty string for primary key. it implies that when a user enforces an index hint, we will NOT use primary storage in the plan.